docs/Makefile

```diff
@@ -0,0 +1,36 @@
+# Makefile for php-src/docs
+# Copyright (c) The PHP Group
+
+# If people set these on the make command line, use 'em
+
+SPHINXBUILD ?= sphinx-build
+
+SOURCEDIR = source
+BUILDDIR = build
+RSTFMT = rstfmt
+RSTFMTFLAGS = -w 80
+
+rwildcard = $(foreach d,$(wildcard $(1:=/*)),$(call rwildcard,$d,$2) $(filter $(subst *,%,$2),$d))
+FILES = $(call rwildcard,$(SOURCEDIR),*.rst)
+
+all : html
+
+.PHONY : check-formatting clean html preflight
+.SUFFIXES : # Disable legacy behavior
+
+check-formatting :
+	$(RSTFMT) $(RSTFMTFLAGS) --check $(SOURCEDIR)
+
+clean :
+	rm -rf -- $(wildcard $(SOURCEDIR)/.~ $(BUILDDIR))
+
+html : preflight
+	$(SPHINXBUILD) -M $@ $(SOURCEDIR) $(BUILDDIR)
+	@printf 'Browse the \e]8;;%s\e\\%s\e]8;;\e\\.\n' \
+		"file://$(abspath $(BUILDDIR))/$@/index.$@" "php-src html docs locally"
+
+preflight : $(SOURCEDIR)/.~
```

Comment:
I don't think building should depend on correct formatting. Some people might not like their code getting formatted on every build, depending on their IDE settings. I'd prefer just keeping formatting a manual command.

Reply:
They might not, however the build will do anyway (our assumption is that CI is parenting here). So the benefit then is to get the result in small changes as fast and often as possible. If some people do not like this, there is no requirement to use the managed build, they remain free to manage their build individually. There are no constraints enforced.

Reply:
And thinking: Looking into https://github.com/php/php-src/pull/16888#issuecomment-2491509834 , the build could benefit from a lint command for the same reasons as given there:

> maybe your CI should run sphinx-lint in addition to rstfmt. I'm quite sure sphinx-lint would have reported the broken literal markup at least.

Reply:
Fair enough. I still don't like forced execution for linters, but we can always remove if somebody complains.